# GROUP ACTIVITY

In this activity, there all 12 questions prepared with varying levels of difficulty. Your task is to answer as many as you can AS A GROUP. Each question is independent of one another. Delegate and strategize on how you will answer the most number of questions. **Good luck! :)**

#### First, setup the data

In [12]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [13]:
schools = pd.read_csv("schools_combined.csv")

C:\Users\Optiplex\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3057: DtypeWarning: Columns (10,11) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [14]:
schools.head()

Unnamed: 0  school_id   school_name                    region  \
0           0     300002  Bacarra NCHS  Region I - Ilocos Region   
1           1     300002  Bacarra NCHS  Region I - Ilocos Region   
2           2     300002  Bacarra NCHS  Region I - Ilocos Region   
3           3     300002  Bacarra NCHS  Region I - Ilocos Region   
4           4     300002  Bacarra NCHS  Region I - Ilocos Region   

       province municipality      division   year_level  gender  enrollment  \
0  Ilocos Norte      Bacarra  Ilocos Norte   first year    male         299   
1  Ilocos Norte      Bacarra  Ilocos Norte   first year  female         250   
2  Ilocos Norte      Bacarra  Ilocos Norte  second year    male         246   
3  Ilocos Norte      Bacarra  Ilocos Norte  second year  female         284   
4  Ilocos Norte      Bacarra  Ilocos Norte   third year    male         220   

      latitude  longitude      ID  Total_Enro  Total_Inst  Rooms_used  \
0  18.24977222  120.61305  300002        1868          55        43.0   
1  18.24977222  120.61305  300002        1868          55        43.0   
2  18.24977222  120.61305  300002        1868          55        43.0   
3  18.24977222  120.61305  300002        1868          55        43.0   
4  18.24977222  120.61305  300002        1868          55        43.0   

   Rooms_unused              Type_of_Sc  
0           0.0  School with no Annexes  
1           0.0  School with no Annexes  
2           0.0  School with no Annexes  
3           0.0  School with no Annexes  
4           0.0  School with no Annexes

In [15]:
schools.sort_values(by="school_id")

Unnamed: 0  school_id                             school_name  \
18792       18792     100002                             Bacarra CES   
18803       18803     100002                             Bacarra CES   
18802       18802     100002                             Bacarra CES   
18801       18801     100002                             Bacarra CES   
18800       18800     100002                             Bacarra CES   
18799       18799     100002                             Bacarra CES   
18793       18793     100002                             Bacarra CES   
18797       18797     100002                             Bacarra CES   
18796       18796     100002                             Bacarra CES   
18795       18795     100002                             Bacarra CES   
18794       18794     100002                             Bacarra CES   
18798       18798     100002                             Bacarra CES   
18815       18815     100003                                Buyon ES   
18814       18814     100003                                Buyon ES   
18813       18813     100003                                Buyon ES   
18812       18812     100003                                Buyon ES   
18811       18811     100003                                Buyon ES   
18810       18810     100003                                Buyon ES   
18809       18809     100003                                Buyon ES   
18808       18808     100003                                Buyon ES   
18807       18807     100003                                Buyon ES   
18806       18806     100003                                Buyon ES   
18805       18805     100003                                Buyon ES   
18804       18804     100003                                Buyon ES   
18827       18827     100004               Ganagan Elementary School   
18826       18826     100004               Ganagan Elementary School   
18825       18825     100004               Ganagan Elementary School   
18824       18824     100004               Ganagan Elementary School   
18823       18823     100004               Ganagan Elementary School   
18822       18822     100004               Ganagan Elementary School   
...           ...        ...                                     ...   
18762       18762     321601                        Tanauan City NHS   
18763       18763     321601                        Tanauan City NHS   
18764       18764     321601                        Tanauan City NHS   
18765       18765     321601                        Tanauan City NHS   
18766       18766     321601                        Tanauan City NHS   
18767       18767     321601                        Tanauan City NHS   
18775       18775     322502                             Tumanan NHS   
18773       18773     322502                             Tumanan NHS   
18772       18772     322502                             Tumanan NHS   
18774       18774     322502                             Tumanan NHS   
18770       18770     322502                             Tumanan NHS   
18769       18769     322502                             Tumanan NHS   
18768       18768     322502                             Tumanan NHS   
18771       18771     322502                             Tumanan NHS   
18783       18783     322503                               Bucto NHS   
18781       18781     322503                               Bucto NHS   
18780       18780     322503                               Bucto NHS   
18782       18782     322503                               Bucto NHS   
18778       18778     322503                               Bucto NHS   
18777       18777     322503                               Bucto NHS   
18776       18776     322503                               Bucto NHS   
18779       18779     322503                               Bucto NHS   
18784       18784     322601  Candon City Information Technology NHS   
18785       18785     322601  Candon City Infor

#### Before proceeding with the questions, check first with your groupmates if you all have the correct dataframe.

1. What is the size of the dataframe?

In [16]:
schools.shape

(159744, 18)

2. What are the columns?

In [17]:
schools.columns

Index(['Unnamed: 0', 'school_id', 'school_name', 'region', 'province',
       'municipality', 'division', 'year_level', 'gender', 'enrollment',
       'latitude', 'longitude', 'ID', 'Total_Enro', 'Total_Inst', 'Rooms_used',
       'Rooms_unused', 'Type_of_Sc'],
      dtype='object')

3. How many schools are there?

In [18]:
schools.school_id.nunique()

14003

4. How many missing values are there?

In [19]:
schools.isna().sum()

Unnamed: 0       0
school_id        0
school_name      0
region           0
province         0
municipality     0
division         0
year_level       0
gender           0
enrollment       0
latitude         0
longitude        0
ID               0
Total_Enro       0
Total_Inst       0
Rooms_used      12
Rooms_unused    12
Type_of_Sc       0
dtype: int64

Alright! Get started!

## Exercise

***Easy***

1. Which has a greater number of overall enrollees: males or females? (use `enrollment` column)

Insert markdown final answer: male

In [63]:
#Insert code here
schools.groupby(["gender"])[["enrollment"]].sum()

enrollment
gender            
female     3870769
male       4053979

2. How many municiapalities are covered by the dataframe?

Insert markdown final answer: 1143

In [64]:
#Insert code here
len(schools[['municipality','province','region']].drop_duplicates())

1143

3. Which year_level has the highest number of enrollees? (use `enrollment` column)

Insert markdown final answer: grade 4

In [65]:
#Insert code here
schools.groupby(["year_level"])[["enrollment"]].sum().sort_values(by = 'enrollment', ascending = False)

enrollment
year_level             
grade 4          920665
grade 5          846376
grade 3          844424
grade 2          824246
first year       817414
grade 1          792898
grade 6          777489
second year      764328
third year       710006
fourth year      626902

4. Are the school_id equal to school_name? What is the discrepancy?

Insert markdown final answer: 1988

In [73]:
#Insert code here
schools.school_id.nunique()-schools.school_name.nunique()

1988

5. How many teachers are there overall?

Insert markdown final answer: 179044

In [79]:
#Insert code here
schools2=schools.drop_duplicates(subset='school_id', keep="first")
schools2['Total_Inst'].sum()

179044

***Medium***

6. Which province has the highest count of schools? How many is it?

Insert markdown final answer: Leyte,1241

In [80]:
#Insert code here
schools3=schools2.groupby(["province"]).size()
schools3.sort_values(ascending=False).head(5)

province
Leyte                1241
Pangasinan            834
North Cotabato        708
Negros Oriental       695
Zamboanga Del Sur     694
dtype: int64

7. How many provinces only has one school? Which provinces are these?

Insert markdown final answer:6,(Abra,Zamboanga Del Norte, Sulu, Apayao, Quirino, Mountain Province)

In [81]:
#Insert code here
new_school = schools.groupby("province")["school_id"].nunique().to_frame().sort_values(by="school_id")

new_school[new_school["school_id"] == 1]

school_id
province                      
Abra                         1
Zamboanga Del Norte          1
Sulu                         1
Apayao                       1
Quirino                      1
Mountain Province            1

8. How many municipalities are there in ARMM?

Insert markdown final answer: 23

In [82]:
#Insert code here
schools[schools["region"] == "ARMM - Autonomous Region in Muslim Mindanao"].groupby("region")["municipality"].nunique().to_frame()

municipality
region                                                   
ARMM - Autonomous Region in Muslim Mindanao            23

9. Which school has the most number of rooms (used + unused)?

Insert markdown final answer: Rizal High School

In [20]:
schools.columns

Index(['Unnamed: 0', 'school_id', 'school_name', 'region', 'province',
       'municipality', 'division', 'year_level', 'gender', 'enrollment',
       'latitude', 'longitude', 'ID', 'Total_Enro', 'Total_Inst', 'Rooms_used',
       'Rooms_unused', 'Type_of_Sc'],
      dtype='object')

In [39]:
#
schools["totalRooms"] = schools["Rooms_used"]+schools["Rooms_unused"]

schools[schools["totalRooms"]==schools["totalRooms"].max()]["school_name"]

16776    Rizal High School
16777    Rizal High School
16778    Rizal High School
16779    Rizal High School
16780    Rizal High School
16781    Rizal High School
16782    Rizal High School
16783    Rizal High School
Name: school_name, dtype: object

10. Within the schools in Central Visayas, how many female enrollees do they have all in all? (use `enrollment` column)

In [27]:
schools[(schools["region"].str.contains("Central Visayas")) & (schools["gender"]=="female")]["enrollment"].sum()

222644

Insert markdown final answer: 222644 female students

***Hard***

11. How many secondary students are there overall? (use `enrollment` column)

In [83]:
secondary = schools[(schools["year_level"] == "first year") | (schools["year_level"] == "second year") | (schools["year_level"] == "third year") | (schools["year_level"] == "fourth year")]
secondary["enrollment"].sum()

2918650

Insert markdown final answer: 2918650

12. Which school_id has the highest teacher-student ratio?

Insert markdown final answer: 121426

In [75]:
schoolsGrouped = schools.groupby("school_id")

In [76]:
(schoolsGrouped["Total_Inst"].max()/schoolsGrouped["enrollment"].sum()).to_frame().sort_values(by=0,ascending=False)

0
school_id          
121426     0.500000
300193     0.394161
319603     0.348548
121793     0.285714
131795     0.272727
132791     0.272727
132804     0.250000
131897     0.250000
122486     0.250000
122725     0.250000
115774     0.250000
122657     0.250000
121501     0.222222
102321     0.202899
113790     0.200000
125660     0.200000
102323     0.200000
122487     0.200000
122477     0.200000
121490     0.200000
121526     0.200000
125350     0.200000
124888     0.200000
109414     0.200000
319506     0.181818
131946     0.181818
102327     0.176471
131867     0.170732
120981     0.166667
121391     0.166667
...             ...
103280     0.000000
121951     0.000000
122474     0.000000
319903     0.000000
317926     0.000000
130342     0.000000
104454     0.000000
131770     0.000000
103862     0.000000
124025     0.000000
130363     0.000000
130365     0.000000
124982     0.000000
111424     0.000000
131710     0.000000
115311     0.000000
317921     0.000000
129835     0.000000
317920     0.000000
115254     0.000000
115256     0.000000
115375     0.000000
317108     0.000000
317105     0.000000
131830     0.000000
316637     0.000000
103148     0.000000
120266     0.000000
120603     0.000000
125270     0.000000

[14003 rows x 1 columns]